### PR TITLE
Make Java auto-detect find sdkman Java installations

### LINKED ISSUE #1630: PolyMC doesn't auto-detect sdkman Java installations
### Role

I'm a contributor

### Suggestion

Make PolyMC find sdkman installations when using the auto-detect feature 

### Benefit

It's easier to use Java versions that are installed via sdkman, and it removes the need for manually selecting them which is somewhat bothersome

### This suggestion is unique

- [X] I have searched the issue tracker and did not find an issue describing my suggestion, especially not one that has been rejected.

### You may use the editor below to elaborate further.

I'll most likely work on this myself, this issue is supposed to be like a tracker.

### PR DESCRIPTION
**Pull request by HeyaGlitz**
Fixes https://github.com/PolyMC/PolyMC/issues/1630

### CODE REVIEW

File: launcher/java/JavaUtils.cpp [hunk 1]
```diff
@@ -448,6 +448,7 @@ QList<QString> JavaUtils::FindJavaPaths()
     scanJavaDir("/opt/jdks");
     // flatpak
     scanJavaDir("/app/jdk");
+    scanJavaDir(FS::PathCombine(std::getenv("HOME"), ".sdkman/candidates/java"));
```
Comment (ghost) [inline launcher/java/JavaUtils.cpp]: You can use QDir::homeDir() instead.

**Review by ghost — CHANGES_REQUESTED** [review]

**Comment by ghost** [comment]
Any reason to use FS::PathCombine rather than std::filesystem::path or string concatenation?

**Comment by HeyaGlitz** [comment]
FS::PathCombine is already previously used in this source file therefore I think it's best to be consistent. Those two methods would work perfectly fine though.

**Comment by ghost** [comment]
Undocumented code using random useless methods🔥🔥🔥

**Review by ghost — APPROVED** [review]

**Pull request merged**